### ISSUE TITLE
Sql Client XML Help file has bad HTML that breaks Sandcastle

### ISSUE THREAD

**Issue by MCFHTAGENTS** [EWSoftware/SHFB#787]
Please can I get your help with the issue here:
[https://github.com/dotnet/SqlClient/issues/450](https://github.com/dotnet/SqlClient/issues/450)
I would like to determine where the `System.Data.SqlClient.xml` that is in my working directory is coming from - the `System.Data.SqlClient `is not referenced by my project directly as I have moved to `Microsoft.Data.SqlClient` and it is not present in C:\Program Files (x86)\Sandcastle\Data\Reflection although `System.Data.xml ` is (and it doesn't contain the th tag).
and the SQL Client team can't find it there end either.

Thanks
Mark

**Comment by EWSoftware** [comment]
What version of SHFB are you using?  If you upgrade to the latest release or at least v2018.5.29.0, it will log invalid comments files with a warning and ignore them and continue with the build.  If you are using older versions of some of the .NET Core/Standard packages, they had a load of bad XML comments.  Newer versions of the packages seem to have resolved most of the issues so updating to a newer release of the NuGet packages may also resolve the issue.

**Comment by MCFHTAGENTS** [comment]
Should be latest of everything. i will confirm later but I do seem to recall 2019.11. I think it is resulting in all the summaries disappearing

**Comment by EWSoftware** [comment]
The file is ignored so the summaries will not display in the resulting help.  You'll need to fix the comments files manually or update to a package that contains the valid ones for the inherited/base comments to display.  The log file will contain the ignored invalid XML comments file names (BE0031 warnings).

**Comment by MCFHTAGENTS** [comment]
Ok so the interesting thing is where does this file come from! I don’t see it in your cached versions Is there a way I can see that in log? It is getting it via reflection I think

**Comment by MCFHTAGENTS** [comment]
Eric
OK I can confirm SFHB version 2019.11.17.0

In terms of the System.Data.SqlClient.xml file there are some 40 copies of this on my PC mainly in the `c:\program files(x86)\dotnet\sdk\NuGetFallbackFolder\system.data.sqlclient\{version}\ref\{platform`}

and my `.nuget\packages` folder - again with various version and platform subdirectories.

As this is a reflection entry I have no idea which one it is gathering in - any way I can get this in the logs?. The log file I have just references the copy it creates in my working directory.

Thanks
Mark

**Comment by EWSoftware** [comment]
Look further back in the log in the "Validating and copying documentation source information" section.  There it should detail which reference assemblies are being used and which XML comments files are being copied to the working folder.  I don't recall offhand if it lists the reference assembly XML comments files in those copied but if you find the reference assembly in the list of references, the comments file will be in the same location as it.

**Comment by MCFHTAGENTS** [comment]
I have log entries like this:
```
References to include:
    BouncyCastle.Crypto
    dotnet-aspnet-codegenerator-design
    EPPlus
    FastMember
    HtmlAgilityPack
    IPAddressRange
    Joonasw.AspNetCore.SecurityHeaders

```
but no paths!

**Comment by MCFHTAGENTS** [comment]
ok - this is curious. It is copying in the XML file from
` {my directory}\.nuget\packages\system.data.sqlclient\4.4.0\ref\netstandard2.0` based on the file date and size (and I can see the rogue <th> in there.

The latest version of the library is 4.8.1 in Nuget however I don't reference this in my project not least as I have switched to Microsoft.Data.SqlClient although the files for the later versions are present in `{my directory}\.nuget\packages\system.data.sqlclient`

I have loaded my library (actually a .NET core 3.1 website) into ILSpy and cannot see a dependency on `System.Data.SqlClient` and this is what I would expect so I am interested you are seeing one.
